DependencyInjection/Compiler/SecurityContextPass.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCacheBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCacheBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * In Symfony < 2.6, replace the new security.token_storage service with the  
+ * deprecated security.context service. 
+ */
+class SecurityContextPass implements CompilerPassInterface
+{
+    const ROLE_PROVIDER_SERVICE = 'fos_http_cache.user_context.role_provider';
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->has(self::ROLE_PROVIDER_SERVICE)) {
+            return;
+        }
+        
+        if (!$container->has('security.token_storage')) {
+            $definition = $container->getDefinition(self::ROLE_PROVIDER_SERVICE);
+            $definition->replaceArgument(0, new Reference('security.context'));
```

Comment:
``security.context`` is also defined outside a firewall. I guess that making it optional is done for the case where the SecurityBundle is not registered. However, does the listener make sense at all when the security system is not there ? If no, the dependency should be required in the class and the service itself should be removed when the security system is not used